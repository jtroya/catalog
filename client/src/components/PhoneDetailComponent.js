import React, { Component } from 'react';
import { Row, Col, Table } from 'react-materialize';

class PhoneDetailComponent extends Component {
  componentDidMount () {
    this.props.getInfoDetail(this.props.phoneId);
  }

  renderInfo = () => {
    const { brand, model, color, description, price, storage, imageUrl  } = this.props.selectedItem[0];
    return (
      <Row className="container">
        <Col s={ 12 } m={ 4 } l={ 4 }>
          <div className="div_panel scale-transition">
            <img className="div_panel-image" alt={ model } src={ imageUrl } />
          </div>
        </Col>
        <Col s={ 12 } m={ 8 } l={ 8 }>
          <Table>
            <tbody>
              <tr><td>Model</td><td>{ model }</td></tr>
              <tr><td>Brand</td><td>{ brand }</td></tr>
              <tr><td>Colour</td><td>{ color }</td></tr>
              <tr><td>Description</td><td>{ description }</td></tr>
              <tr><td>Price</td><td>{ price }</td></tr>
              <tr><td>Storage</td><td>{ storage }</td></tr>
            </tbody>
          </Table>
        </Col>
      </Row>
    );
  }

  render() {    
    return (
      <Row>
        { this.props.selectedItem[0] &&
          this.renderInfo()
        }
      </Row>
    )
  }
}

export default PhoneDetailComponent;
