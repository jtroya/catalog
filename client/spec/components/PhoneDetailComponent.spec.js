import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PhoneDetailComponent from '../../src/components/PhoneDetailComponent';

describe('PhoneDetailComponent', () => {
  let wrapper;
  const getInfoDetail = jest.fn();
  const fakeSelectedItem = [{ 
    model: 'my model', 
    brand: 'my brand', 
    color: 'red',  
    description: 'my description', 
    price: 123,
    storage: '16 GB',
    imageUrl: 'img-url'
  }];

  beforeEach(() => {
    wrapper = shallow(<PhoneDetailComponent 
      phoneId={ 2 }
      selectedItem={ fakeSelectedItem }
      getInfoDetail={ getInfoDetail }
    />);
  });

  it('should render the component', () => {
    expect(wrapper.find('PhoneDetailComponent')).toBeTruthy();
  });

  it('should not render if there is no data', () => {
    wrapper.setProps({ selectedItem: [] });
    expect(wrapper.find('Row')).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const tree = shallow(<PhoneDetailComponent 
      phoneId={ 2 }
      selectedItem={ fakeSelectedItem }
      getInfoDetail={ getInfoDetail }
    />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
