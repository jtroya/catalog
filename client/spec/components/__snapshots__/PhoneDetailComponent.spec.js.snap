// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneDetailComponent matches the snapshot 1`] = `
<Row>
  <Row
    className="container"
  >
    <Col
      l={4}
      m={4}
      s={12}
    >
      <div
        className="div_panel scale-transition"
      >
        <img
          className="div_panel-image"
        />
      </div>
    </Col>
    <Col
      l={8}
      m={8}
      s={12}
    >
      <Table>
        <tbody>
          <tr>
            <td>
              Model
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Brand
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Colour
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Description
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Price
            </td>
            <td />
          </tr>
          <tr>
            <td>
              Storage
            </td>
            <td />
          </tr>
        </tbody>
      </Table>
    </Col>
  </Row>
</Row>
`;
